### ISSUE TITLE
Bugs

### ISSUE THREAD

**Issue by bernardmizzi** [Cadrach/lol_ise#62]
Are there any current bugs that you know of and need a fix? I'm asking you this so that I can write properties so that in my runnable version they won't occur.

**Comment by Cadrach** [comment]
None that I know of.

I would be interested in reading your result if I may once you are finished.

**Comment by bernardmizzi** [comment]
Of course, hopefully next June :)

**Comment by bernardmizzi** [comment]
Mate, is there a way how to make the functions defined in the controller in viewMain.js avaliable also in the window object?

**Comment by Cadrach** [comment]
You might try something like `window.functionName = $scope.functionName` but the result has a good change of being wonky.

**Comment by bernardmizzi** [comment]
would it make a difference if instead of $scope.functionName I wrote $window.functionName? If I'm not mistaken, $window refers to the window object, right?

**Comment by Cadrach** [comment]
Yes, I think this has good changes of breaking the tool. You need to understand how Angular scopes work, you may start here you want: https://docs.angularjs.org/guide/scope

**Comment by bernardmizzi** [comment]
I'll have a look at that thankyou. Before, I was trying the tutorial shown in the following link: https://coderwall.com/p/smkwhg/access-scope-outside-of-angularjs#comment_29726. But, when I try to access the newly retrieved $scope and apply $apply I get 'Cannot read property '$apply' of undefined'.

**Comment by Cadrach** [comment]
maybe you are doing it too early (this element does not exist yet) or on the wrong element (the element you choose does not have any scope)

**Comment by bernardmizzi** [comment]
One last question, is the angular controller sort of instantiated at any time or it's just there and can be used when you want?

**Comment by Cadrach** [comment]
It is instanciated and not available at anytime, but probably not long after the page has finished loading.

**Comment by bernardmizzi** [comment]
And I assume that nowhere in your code you're instantiating it right? It is sort of done automatically I assume.

**Comment by Cadrach** [comment]
Yes, it is done through the routing

**Comment by bernardmizzi** [comment]
Do you have any idea of how I can modify the library to not use routing?

**Comment by Cadrach** [comment]
Really no, Angular is the framework used for the tool, so if you want to remove the routing it is like trying to remove the framework. So short of rewriting the tool from scratch with no framework I do not see how you can do it.

**Comment by bernardmizzi** [comment]
So I sort of figured that part out, another question is where are the local functions inside the controller stored? When we define a function using $scope.functionName, function functionName can be found in $scope, but where are those without $scope found?

**Comment by Cadrach** [comment]
Well, by definition they cannot live outside the scope, their meaning and functionality require a scope to work
